build-source.sh

```diff
@@ -113,7 +113,22 @@ rm -rf tmp || true
 
 if [ -n "$CHANGE_ID" ]; then
   # This is a PR. Publish each PR for each project into its own repository
-  GIT_REPO_NAME="$(basename "${GIT_URL%.git}")"
+
+  if [ -z "$GIT_URL" ]; then
+    GIT_URL=$GIT_URL_1
+  fi
+  if [ -z "$GIT_URL" ]; then
+    GIT_URL=$( (cd source && git remote get-url origin) || true)
+  fi
+
+  if [ -n "$GIT_URL" ]; then
+    GIT_REPO_NAME="$(basename "${GIT_URL%.git}")"
+  else
+    echo "Cannot determine git repo name. Try to use the job name instead." \
+         "May produce incorrect apt repository name."
```

Comment:
The output of this would be `job name instead.May produce incorrect`, wouldn't it?

Reply:
No, because that's 2 arguments to `echo`. `echo` inserts whitespace between arguments.